modules/router-store/src/serializers/minimal_serializer.ts

```diff
@@ -0,0 +1,33 @@
+import { RouterStateSnapshot, ActivatedRouteSnapshot } from '@angular/router';
+import { BaseRouterStoreState, RouterStateSerializer } from './shared';
+
+export interface MinimalRouteSnapshot {
+  params: RouterStateSnapshot['root']['params'];
+  url: RouterStateSnapshot['root']['url'];
+  queryParams: RouterStateSnapshot['root']['queryParams'];
+  data: RouterStateSnapshot['root']['data'];
+}
+
+export interface MinimalRouterStateSnapshot extends BaseRouterStoreState {
+  root: MinimalRouteSnapshot;
+  url: string;
+}
+
+export class MinimalRouterStateSerializer
```

Comment:
After taking a second look at this, I think this should match what the custom serializer does in the docs. If we just give them the root snapshot, the params will be empty, the data will be empty, only leaving the url and global queryParams.

Reply:
That would make more sense

Reply:
I've pushed these changes. One caveat is that users won't be able to access params from higher up. Another implementation I've encountered a few times is to loop over all of the children and pluck the params from each child and build up an object with all of the params.